external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -200,16 +215,22 @@ private void setAcked(TopicPartition tp, long fetchOffset) {
     @Override
     public void nextTuple() {
         if (initialized) {
-            if (commit()) {
-                commitOffsetsForAckedTuples();
-            }
+            try {
+                refreshPartitionIfNeeded();
 
-            if (poll()) {
-                setWaitingToEmit(pollKafkaBroker());
-            }
+                if (commit()) {
+                    commitOffsetsForAckedTuples();
+                }
 
-            if (waitingToEmit()) {
-                emit();
+                if (poll()) {
+                    setWaitingToEmit(pollKafkaBroker());
+                }
+
+                if (waitingToEmit()) {
+                    emit();
+                }
+            } catch (Exception e) {
```

Comment:
It's not uncommon to encounter an exception when kafka consumer poll data from broker and I don't think we should restart it. But I should not catch the whole function. This has been resolved in the new patch.